reporter/src/main/kotlin/reporters/spdx/SpdxDocumentModelMapper.kt

```diff
@@ -216,14 +216,29 @@ private fun getSpdxCopyrightText(
 }
 
 private fun Package.toSpdxExternalReferences(): List<SpdxExternalReference> {
-    if (purl.isEmpty()) return emptyList()
+    // var externalRefs: List<SpdxExternalReference> = emptyList()
+    val externalRefs = mutableListOf<SpdxExternalReference>()
 
-    val reference = SpdxExternalReference(
-        referenceType = SpdxExternalReference.Type.Purl,
-        referenceLocator = purl
-    )
+    if (purl.isNotEmpty()) {
+        externalRefs += SpdxExternalReference(
+            referenceType = SpdxExternalReference.Type.Purl,
+            referenceLocator = purl
+        )
+    }
+
+    if (!cpe.isNullOrEmpty()) {
+        val referenceType = if (cpe!!.startsWith("cpe:2.3")) {
```

Comment:
Ok, I realize now that smart casts don't work here, but here's some syntactic sugar to work around that:

    cpe?.takeUnless { it.isEmpty() }?.let {
        val referenceType = if (it.startsWith("cpe:2.3")) {
            SpdxExternalReference.Type.Cpe23Type
        } else {
            SpdxExternalReference.Type.Cpe22Type
        }

        externalRefs += SpdxExternalReference(
            referenceType,
            referenceLocator = it
        )
    }

Reply:
In this line, like in my code snippet, use `it` instead of `cpe!!`.